app/script/view_model/content/MessageListViewModel.js

```diff
@@ -380,20 +372,34 @@ z.viewModel.content.MessageListViewModel = class MessageListViewModel {
    * Scroll to given message in the list.
    *
    * @note Ideally message is centered horizontally
-   * @param {z.entity.Message} message_et - Target message
+   * @param {string} messageId - Target message's id
    * @returns {undefined} No return value
    */
-  _focus_message(message_et) {
-    const message_element = $(`.message[data-uie-uid="${message_et.id}"]`);
-
-    if (message_element.length) {
-      const message_list_element = this._getMessagesContainer();
-      message_list_element.scrollBy(message_element.offset().top - message_list_element.height() / 2);
-    }
-  }
-
-  scroll_height(change_in_height) {
-    this._getMessagesContainer().scrollBy(change_in_height);
+  _focusMessage(messageId) {
+    const messageIsLoaded = !!this.conversation().getMessage(messageId);
+    const conversationEntity = this.conversation();
+
+    const loadMessagePromise = messageIsLoaded
+      ? Promise.resolve()
+      : this.conversation_repository
+          .get_message_in_conversation_by_id(conversationEntity, messageId)
+          .then(messageEntity => {
+            conversationEntity.remove_messages();
+            return this.conversation_repository.getMessagesWithOffset(conversationEntity, messageEntity, 30);
+          });
+
+    loadMessagePromise.then(() => {
+      z.util.afterRender(() => {
+        const message_element = $(`.message[data-uie-uid="${messageId}"]`);
+
+        if (message_element.length) {
+          message_element.removeClass('message-marked');
+          const message_list_element = this._getMessagesContainer();
+          message_list_element.scrollBy(message_element.offset().top - message_list_element.height() / 2);
```

Comment:
You might want to look into https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView for this.

Reply:
Good idea but unfortunately, this scrolls the whole app :D